lib/trento/application/integration/discovery/policies/sap_system_policy.ex

```diff
@@ -20,235 +29,130 @@ defmodule Trento.Integration.Discovery.SapSystemPolicy do
         "agent_id" => agent_id,
         "payload" => payload
       }) do
-    payload
-    |> Enum.flat_map(fn sap_system -> parse_sap_system(sap_system, agent_id) end)
-    |> Enum.reduce_while(
-      {:ok, []},
-      fn
-        {:ok, command}, {:ok, commands} -> {:cont, {:ok, commands ++ [command]}}
-        {:error, _} = error, _ -> {:halt, error}
-      end
-    )
+    case SapSystemDiscoveryPayload.from_list(payload) do
+      {:ok, sap_systems} ->
+        sap_systems
+        |> Enum.flat_map(fn sap_system -> build_commands(sap_system, agent_id) end)
+        |> Enum.reduce_while(
+          {:ok, []},
+          fn
+            {:ok, command}, {:ok, commands} -> {:cont, {:ok, commands ++ [command]}}
+            {:error, _} = error, _ -> {:halt, error}
+          end
+        )
```

Comment:
Couldn't you just `Enum.filter` out `{:error, _}` tuples?

Reply:
This piece of code actually was there, done by @fabriziosestito . I didn't even consider that I could do it better than he did hehe

Reply:
Fair enough :+1: